Tribler/Main/Utility/GuiDBHandler.py

```diff
@@ -165,41 +156,34 @@ def wrapper():
 
         wrapper.__name__ = str(name)
 
-        if not self.onSameThread(workerType) or delay:
-            if workerType == "dbThread":
-                task_name = uId
-                if not task_name:
-                    self._auto_counter += 1
-                    task_name = "guidbhandler %d" % self._auto_counter
+        db_thread_wanted = workerType == "dbThread"
+        threadpool_wanted = workerType == "guiTaskQueue"
+        can_run_in_same_thread = (db_thread_wanted and isInIOThread()) or (threadpool_wanted and isInThreadPool())
```

Comment:
Oh! so onSameThread meant "we are going to call the stuff on the same thread" as opposed to "we are on the same thread we want to schedule the task in"?

So if we are on the reactor thread, we disregard the request to be executed on the threadpool as the only thing it wants really is to run out of the GUI thread?